### PR TITLE
Added Traefik Reverse Proxy and Loadbalancer

### LINKED ISSUE #68: Add proper https certificate
Start with a manual approach, and later use certbot to refresh it automaticly if we have time

### PR DESCRIPTION
**Pull request by jlndk**
Solves #68.

To ensure fresh HTTPS certificates this PR adds a reverse proxy / loadbalancer in front of our app, that automaticly handles it. Besides HTTPS certificate renewals the proxy also allows us to scale our service easier in the future, since it also automaticly load balances the trafic between multiple instances of the same container, and also allows us to split our service into more containers and define routing on the TCP or HTTP level (based on port, host, http verb, headers, etc.).

Traefik also has inbuild statistics and monitoring which also might come in handy later

### CODE REVIEW

**Comment by jlndk** [comment]
Before we merge we should test https certificate generation on the server, as i cannot do more testing locally without the domain.

**Comment by jlndk** [comment]
Also the traefik dashboard/api is still publicly accessible without authentication.
Before we merge this PR this should also be resolved.

File: docker-compose.prod.yml [hunk 1]
```diff
@@ -18,10 +44,33 @@ services:
...
         volumes:
             - ./aspnetkeys:/root/.aspnet/DataProtection-Keys
+        labels:
+            # Tell traefik that this container should be exposed to the public
+            - "traefik.enable=true"
+            # Tell traefik which port our container listens on
+            - "traefik.http.services.webapp.loadbalancer.server.port=80"
+
+            # Add http entrypoint (Tell where the service should be available)
+            - "traefik.http.routers.webapp.rule=Host(`minitwit.tk`) || Path(`/{path:.*}`)"
```
Comment (tobloef) [inline docker-compose.prod.yml:61]: Could we use a yaml reference here to only have the domain defined in one place?

**Review by tobloef — DISMISSED** [review]

File: docker-compose.prod.yml (hunk 1 above)
Comment (Hjaltesorgenfrei) [reply docker-compose.prod.yml:61]: No

**Review by Hjaltesorgenfrei — APPROVED** [review]
:+1:

**Pull request merged**